etna/transforms/timestamp/holiday.py

```diff
@@ -83,14 +141,32 @@ def _transform(self, df: pd.DataFrame) -> pd.DataFrame:
         -------
         :
             pd.DataFrame with added holidays
+
+        Raises
+        ------
+        ValueError:
+            if the frequency is greater than daily and this is a ``binary`` or ``categorical`` mode
+        ValueError:
+            if the frequency is less than daily and this is ``days_count`` mode
         """
-        if (df.index[1] - df.index[0]) > datetime.timedelta(days=1):
-            raise ValueError("Frequency of data should be no more than daily.")
+        if bigger_than_day(self.freq) and self._mode is not HolidayTransformMode.days_count:
+            raise ValueError("For binary and category modes frequency of data should be no more than daily.")
 
         cols = df.columns.get_level_values("segment").unique()
         out_column = self._get_column_name()
 
-        if self._mode is HolidayTransformMode.category:
+        if self._mode is HolidayTransformMode.days_count:
+            numbers_part = "".join(itertools.takewhile(str.isdigit, self.freq))
```

Comment:
I think we should utilize smth like `pd.tseries.frequencies.to_offset(ts.freq)` here instead of this code.